File: exercises/r/02_limpieza_superstore.R
# =====================================================
# Ejercicio 2
# Limpieza y preparación de datos - Superstore
# =====================================================


library(readr)
library(dplyr)


# Cargar dataset

ventas <- read_csv(
  "datasets/superstore.csv",
  show_col_types = FALSE
)


# -----------------------------------------
# 1. Valores faltantes
# -----------------------------------------

cat("\nValores faltantes por columna\n")

print(
  colSums(is.na(ventas))
)


# -----------------------------------------
# 2. Duplicados
# -----------------------------------------

cat("\nCantidad de duplicados\n")

print(
  sum(duplicated(ventas))
)


# -----------------------------------------
# 3. Convertir fechas
# -----------------------------------------

ventas <- ventas %>%
  mutate(
    `Order Date` = as.Date(
      `Order Date`,
      format="%m/%d/%Y"
    ),
    
    `Ship Date` = as.Date(
      `Ship Date`,
      format="%m/%d/%Y"
    )
  )


cat("\nTipos de fecha\n")

print(
  class(ventas$`Order Date`)
)


# -----------------------------------------
# 4. Crear días de entrega
# -----------------------------------------

ventas <- ventas %>%
  mutate(
    Delivery_Days =
      as.numeric(
        `Ship Date` - `Order Date`
      )
  )


cat("\nEjemplo nueva variable\n")

print(
  ventas %>%
    select(
      `Order Date`,
      `Ship Date`,
      Delivery_Days
    ) %>%
    head()
)


# -----------------------------------------
# 5. Promedio días entrega
# -----------------------------------------

cat("\nPromedio días de entrega\n")

print(
  mean(
    ventas$Delivery_Days
  )
)

File: exercises/python/02_limpieza_superstore.py
# =====================================================
# Ejercicio 2
# Limpieza y preparación de datos - Superstore
# =====================================================

import pandas as pd


# 1. Cargar dataset

ventas = pd.read_csv(
    "datasets/superstore.csv",
    encoding="latin-1"
)


# 2. Revisar valores faltantes

print("\nValores faltantes:")
print(ventas.isnull().sum())


# 3. Revisar duplicados

print("\nRegistros duplicados:")
print(ventas.duplicated().sum())


# 4. Convertir fechas

ventas["Order Date"] = pd.to_datetime(
    ventas["Order Date"]
)

ventas["Ship Date"] = pd.to_datetime(
    ventas["Ship Date"]
)


print("\nTipos de datos:")
print(
    ventas[
        ["Order Date", "Ship Date"]
    ].dtypes
)


# 5. Crear nueva variable:
# días de entrega

ventas["Delivery Days"] = (
    ventas["Ship Date"]
    -
    ventas["Order Date"]
).dt.days


print("\nEjemplo de días de entrega:")

print(
    ventas[
        [
            "Order Date",
            "Ship Date",
            "Delivery Days"
        ]
    ].head()
)


# 6. Promedio de entrega

print("\nPromedio días de entrega:")

print(
    ventas["Delivery Days"].mean()
)